packages/gatsby-plugin-sass/src/gatsby-node.js

```diff
@@ -75,3 +75,124 @@ exports.onCreateWebpackConfig = (
     },
   })
 }
+
+if (process.env.GATSBY_EXPERIMENTAL_PLUGIN_OPTION_VALIDATION) {
+  const MATH_ALL_KEYS = /^/
+  exports.pluginOptionsSchema = function ({ Joi }) {
+    return Joi.object({
+      implementation: Joi.object({})
+        .unknown(true)
+        .description(
+          `By default the node implementation of Sass (node-sass) is used. To use the implementation written in Dart (dart-sass), you can install sass instead of node-sass and pass it into the options as the implementation`
+        ),
+      postCssPlugins: Joi.array()
+        .items(Joi.object({}).unknown(true))
+        .description(`An array of postCss plugins`),
+      sassRuleTest: Joi.object()
+        .instance(RegExp)
+        .description(`Override the file regex for SASS`),
+      sassRuleModulesTest: Joi.object()
+        .instance(RegExp)
+        .description(`Override the file regex for SASS`),
+      useResolveUrlLoader: Joi.alternatives().try(
+        Joi.boolean(),
+        Joi.object({}).unknown(true)
```

Comment:
We should type this better in the future, but it's not important enough for this v1 👍

Reply:
Yeah I totally agree with you :/ . I found this plugin very tough compared to some others 😅